src/Sylius/Bundle/CoreBundle/Security/UserPasswordResetter.php

```diff
@@ -27,12 +27,17 @@ public function __construct(
     ) {
     }
 
+    /**
+     * @throws UserNotFoundException
+     */
     public function reset(string $token, string $password): void
     {
         /** @var UserInterface|null $user */
         $user = $this->userRepository->findOneBy(['passwordResetToken' => $token]);
 
-        Assert::notNull($user, 'No user found with reset token: ' . $token);
+        if (null === $user) {
+            throw new UserNotFoundException(message: sprintf('No user found with reset token: %s', $token));
```

Comment:
I wonder if giving such explicit feedback is needed, wdyt?
```suggestion
            throw new UserNotFoundException(message: 'Invalid token');
```

Reply:
I don't have any strong opinion, for me messages should be descriptive, but hide unnecessary details. Your suggestion meets that criteria, TBH I wasn't thinking about messages at all. Let's wait for more opinions 🤔. Maybe @NoResponseMate :D.

Reply:
IMO `Invalid token` or `Token not found` would suffice

Reply:
Changed 💃🏼